src/BugsnagUnity/Payload/Metadata.cs

```diff
@@ -1,8 +1,26 @@
 ﻿using System.Collections.Generic;
+using BugsnagUnity;
 
 namespace BugsnagUnity.Payload
 {
   public class Metadata : Dictionary<string, object>, IFilterable
   {
+    private INativeClient NativeClient;
+
+    internal Metadata(INativeClient nativeClient) {
+      NativeClient = nativeClient;
+    }
+
+    public void Add(string section, object newValue) {
+      base.Add(section, newValue);
```

Comment:
This will crash if the key already exists in the dictionary, due to how [dictionaries are designed in C#](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.idictionary-2.add?view=netframework-4.8#System_Collections_Generic_IDictionary_2_Add__0__1_)

Reply:
Correct, and it should since this class is a subclass of Dictionary. I intentionally didn't change this behavior.